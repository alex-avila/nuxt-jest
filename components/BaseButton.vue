<template>
  <button
    :class="withoutStyle ? 'has-no-style' : 'button is-primary'"
    v-bind="$attrs"
    v-on="$listeners"
  >
    <slot />
  </button>
</template>

<script>
export default {
  props: {
    withoutStyle: {
      type: Boolean,
      default: false
    }
  }
}
</script>


<style lang="sass" scoped>
.has-no-style
  color: inherit
  background: none
  border: none
  font: inherit
  cursor: inherit
</style>
