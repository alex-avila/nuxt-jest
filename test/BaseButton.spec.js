import { shallowMount } from '@vue/test-utils'
import BaseButton from '@/components/BaseButton.vue'

describe('BaseButton', () => {
  const buttonContent = 'Add Note'
  const wrapper = shallowMount(BaseButton, {
    slots: {
      default: buttonContent
    }
  })

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  test('displays button content', () => {
    expect(wrapper.text()).toBe(buttonContent)
  })
})
