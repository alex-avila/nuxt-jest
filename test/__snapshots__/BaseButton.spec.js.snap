// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton renders correctly 1`] = `
<button
  class="button is-primary"
>
  Add Note
</button>
`;
